configure.ac

```diff
@@ -80,7 +80,19 @@ AC_CHECK_FUNC([memset], [], [AC_MSG_ERROR([NCCL OFI Plugin requires memset funct
 AC_CHECK_FUNC([realpath], [], [AC_MSG_ERROR([NCCL OFI Plugin requires realpath function.])])
 AC_CHECK_FUNCS([gettid])
 AC_SEARCH_LIBS([dlopen], [dl], [], [AC_MSG_ERROR([NCCL OFI Plugin requires dlopen])])
+
+dnl it is insufficient to just link against pthread with -lpthread in 2024, it's
+dnl modern compilers and linkers prefer top-level -pthread flags to be passed.
+dnl This impacts the way that libc behaves. We need all three.
+dnl
+dnl background:
+dnl   https://stackoverflow.com/a/23251828
+dnl
+CFLAGS="${CFLAGS} -pthread"
+CPPFLAGS="${CPPFLAGS} -pthread"
```

Comment:
CXXFLAGS.  It should not be passed for CPPFLAGS.

Reply:
I think it's actually the other way around, the CFLAGS line is the one that needs to be removed.

It's a preprocessor option: https://gcc.gnu.org/onlinedocs/gcc/Preprocessor-Options.html

> `-pthread`: Define additional macros required for using the POSIX threads library. You should use this option consistently for both compilation and linking. This option is supported on GNU/Linux targets, most other Unix derivatives, and also on x86 Cygwin and MinGW targets.

It's also a linker option: https://gcc.gnu.org/onlinedocs/gcc/Link-Options.html#index-pthread-1

> `-pthread`: Link with the POSIX threads library. This option is supported on GNU/Linux targets, most other Unix derivatives, and also on x86 Cygwin and MinGW targets. On some targets this option also sets flags for the preprocessor, so it should be used consistently for both compilation and linking.

Reply:
i'm fine with that.